Formula/ipopt.rb

```diff
@@ -0,0 +1,47 @@
+class Ipopt < Formula
+  desc "Large-scale nonlinear optimization package"
+  homepage "https://projects.coin-or.org/Ipopt"
+  url "https://www.coin-or.org/download/source/Ipopt/Ipopt-3.12.9.tgz"
+  sha256 "8ff3fe1a8560896fc5559839a87c2530cac4ed231b0806e487bfd3cf2d294ab8"
+  head "https://projects.coin-or.org/svn/Ipopt/trunk", :using => :svn
+
+  option "without-test", "Skip build-time tests (not recommended)"
+
+  depends_on "ampl-mp"
+  depends_on "gcc"
+  # IPOPT is not able to use parallel MUMPS.
+  depends_on "mumps" => "without-mpi"
```

Comment:
mumps is not currently in core

Reply:
There is a [PR for mumps](https://github.com/Homebrew/homebrew-core/pull/22368), and I hope it's merged before this.

Reply:
If `ipopt` needs MPI-less mumps, then maybe the mumps formula should be made MPI-less by default. We do not allow formulas in core to depend on other formulas with non-default options (to limit option conflicts and builds from source).

Reply:
Done by https://github.com/Homebrew/homebrew-core/pull/22368/commits/8a84949a3eb51fdf1856c4eba9dc091846b2bad6